### PR TITLE
{compiler,tools}[GCCcore/13.2.0] Clang v17.0.6

### PR DESCRIPTION
**Pull request by bartoldeman**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by bartoldeman** [comment]
@boegelbot please test @ generoso

**Comment by bartoldeman** [comment]
@boegelbot please test @ jsc-zen2

**Comment by bedroge** [comment]
@boegelbot please test @ jsc-zen3

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by bartoldeman** [comment]
Note that `enable_rtti = False` disables both RTTI and exception handling in the easyblock:
```
    if self.cfg["enable_rtti"]:
            self.cfg.update('configopts', '-DLLVM_REQUIRES_RTTI=ON')
            self.cfg.update('configopts', '-DLLVM_ENABLE_RTTI=ON')
            self.cfg.update('configopts', '-DLLVM_ENABLE_EH=ON')
```
but for flang we only need to skip `'-DLLVM_ENABLE_EH=ON'` (see e.g. the spack package: https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/llvm/package.py) and can keep RTTI without exception handling.

this would need a change to the easyblock, either
a) a new option `"enable_eh"` seperate from `"enable_rtti"`
or
b) like Spack:
```
    if self.cfg["enable_rtti"]:
        self.cfg.update('configopts', '-DLLVM_REQUIRES_RTTI=ON')
        self.cfg.update('configopts', '-DLLVM_ENABLE_RTTI=ON')
        if 'flang' not in self.cfg['llvm_projects'] or version < '15':
            self.cfg.update('configopts', '-DLLVM_ENABLE_EH=ON')
```
@ocaisa what do you think as author of that part of the easyblock?

**Comment by ocaisa** [comment]
@bartoldeman if it works, I'm fine with it. I only disabled those to get a functional build, I have no real understanding of the impact

**Comment by lexming** [comment]
Changes in https://github.com/easybuilders/easybuild-easyconfigs/pull/20050 are relevant to this PR.

File: easybuild/easyconfigs/z/Z3/Z3-4.12.5-GCCcore-13.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+easyblock = 'CMakeMake'
```
Comment (Micket) [inline easybuild/easyconfigs/z/Z3/Z3-4.12.5-GCCcore-13.2.0.eb:1]: rebase without this ec

**Comment by boegel** [comment]
> Note that `enable_rtti = False` disables both RTTI and exception handling in the easyblock:
> 
> ```
>     if self.cfg["enable_rtti"]:
>             self.cfg.update('configopts', '-DLLVM_REQUIRES_RTTI=ON')
>             self.cfg.update('configopts', '-DLLVM_ENABLE_RTTI=ON')
>             self.cfg.update('configopts', '-DLLVM_ENABLE_EH=ON')
> ```
> 
> but for flang we only need to skip `'-DLLVM_ENABLE_EH=ON'` (see e.g. the spack package: https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/llvm/package.py) and can keep RTTI without exception handling.
> 
> this would need a change to the easyblock, either a) a new option `"enable_eh"` seperate from `"enable_rtti"` or b) like Spack:
> 
> ```
>     if self.cfg["enable_rtti"]:
>         self.cfg.update('configopts', '-DLLVM_REQUIRES_RTTI=ON')
>         self.cfg.update('configopts', '-DLLVM_ENABLE_RTTI=ON')
>         if 'flang' not in self.cfg['llvm_projects'] or version < '15':
>             self.cfg.update('configopts', '-DLLVM_ENABLE_EH=ON')
> ```
> 
> @ocaisa what do you think as author of that part of the easyblock?

@bartoldeman Are you up for opening a PR for this?

File: easybuild/easyconfigs/z/Z3/Z3-4.12.5-GCCcore-13.2.0.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/z/Z3/Z3-4.12.5-GCCcore-13.2.0.eb]: fixed in 1c9177d8edfa006b4502d0bbd1311dcf8f60c361

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen3
CORE_CNT=16

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3117.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/boegel/b1bf4dce3f374209d53849eb6954ba93 for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @bartoldeman!

**Pull request merged**